tokio-trace/tokio-trace-core/src/metadata.rs

```diff
@@ -217,10 +245,42 @@ impl<'a> fmt::Debug for Metadata<'a> {
             .field("file", &self.file)
             .field("line", &self.line)
             .field("field_names", &self.fields)
+            .field("callsite_kind", &self.callsite_kind)
             .finish()
     }
 }
 
+#[repr(usize)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+enum CallsiteKindInner {
+    Event,
+    Span,
+}
+
+impl CallsiteKind {
+    /// `Event` callsite
+    pub const EVENT: CallsiteKind = CallsiteKind(CallsiteKindInner::Event);
+
+    /// `Span` callsite
+    pub const SPAN: CallsiteKind = CallsiteKind(CallsiteKindInner::Span);
+
+    pub fn is_span(&self) -> bool {
+        if let CallsiteKind(CallsiteKindInner::Span) = self {
```

Comment:
TIOLI: This could also be expressed as
```rust
match self {
    CallsiteKind(CallsiteKindInner::Span) => true,
    _=> false
}
```
which is one line shorter. Either is fine though.